sec_certs/sample/fips.py

```diff
@@ -420,354 +489,126 @@ def pandas_tuple(self) -> Tuple:
-            for found in self.pdf_data.keywords["fips_certlike"]["Certlike"]:
-                match_in_found = reg_to_match.search(found)
-                match_in_cert = reg_to_match.search(cert)
-                if (
-                    match_in_found is not None
-                    and match_in_cert is not None
-                    and match_in_found.group("id") == match_in_cert.group("id")
-                ):
-                    to_pop.add(cert)
-
-        this_id = int("".join(filter(str.isdigit, cert)))
-
-        for algo in self.heuristics.algorithms:
-            try:
-                algo_id = int("".join(filter(str.isdigit, algo.cert_id)))
-                if algo_id == this_id:
-                    to_pop.add(cert)
-            except ValueError:
-                continue
+                logger.warning(f"Error when parsing tables from {cert.dgst}: {e}")
+                cert.state.policy_extract_ok = False
 
-    def clean_cert_ids(self) -> None:
+    def prune_referenced_cert_ids(self) -> None:
         """
-        Removes algorithm mentions from the cert_id rule matches and stores them into clean_cert_id matches.
+        This method goes through all IDs (numbers) that correspond to FIPS Certificates and are stored in
+        pdf_data.keywords or web_data.mentioned_certs. It performs prunning of these attributes and fills attributes
+        heuristics.prunned_module_references and heuristics.prunned_policy_references. These variables are further
+        processed and Reference objects are created from them.
         """
-        self.state.file_status = True
-        if not self.pdf_data.keywords:
-            return
-
-        if "Cert" not in self.pdf_data.keywords["fips_cert_id"]:
-            self.pdf_data.clean_cert_ids = {}
-            return
-
-        matches = copy.deepcopy(self.pdf_data.keywords["fips_cert_id"]["Cert"])
-
-        alg_set: Set[str] = set()
-        if self.web_data.algorithms is None:
-            raise RuntimeError(f"Algorithms were not found for cert {self.cert_id} - this should not be happening.")
+        html_module_ids = set(self.web_data.mentioned_certs.keys()) if self.web_data.mentioned_certs else set()
+        self.heuristics.module_prunned_references = self._prune_reference_ids_variable(html_module_ids)
 
-        for algo in self.web_data.algorithms:
-            alg_set.add(algo.cert_id)
-
-        for cert_rule in fips_rules["fips_cert_id"]["Cert"]:
-            to_pop = set()
-            for cert in matches:
-                if cert in alg_set:
-                    to_pop.add(cert)
-                    continue
-                self._process_to_pop(cert_rule, cert, to_pop)
-
-            for r in to_pop:
-                matches.pop(r, None)
-
-        matches.pop("#" + str(self.cert_id), None)
-        self.pdf_data.clean_cert_ids = matches
+        if self.pdf_data.keywords:
+            pdf_policy_ids = set(self.pdf_data.keywords["fips_cert_id"].get("Cert", dict().keys()))
```

Comment:
The second argument to `get` being `dict().keys()` is very weird, what is it supposed to be?

Reply:
A bug. This should be `get("cert", dict()).keys()))` Thanks!